### PR TITLE
add links to standard errors in toric folder

### PR DESCRIPTION
**Pull request by fchapoton**
adding links to standard python errors in the doc inside the `schemes/toric` folder

plus a few pep8 cleanups

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/schemes/toric/variety.py [hunk 1]
```diff
@@ -1216,7 +1217,7 @@ def dimension_singularities(self):
         this method will return the dimension of the
         largest-dimensional component.
 
-        Returns -1 if the toric variety is smooth.
+        This returns -1 if the toric variety is smooth.
```
Comment (mkoeppe) [inline src/sage/schemes/toric/variety.py]:
```suggestion
        This returns `-1` if the toric variety is smooth.
```

File: src/sage/schemes/toric/weierstrass.py [hunk 2]
```diff
@@ -176,8 +176,8 @@ def Discriminant(polynomial, variables=None):
         sage: Discriminant([quadratic1, quadratic2])
         -1/16
     """
-    (f, g) = WeierstrassForm(polynomial, variables)
-    return 4*f**3+27*g**2
+    f, g = WeierstrassForm(polynomial, variables)
+    return 4 * f**3 + 27 * g**2
```
Comment (mkoeppe) [inline src/sage/schemes/toric/weierstrass.py]:
```suggestion
    return 4*f**3 + 27*g**2
```
whitespace around top-level arith operators only please

File: src/sage/schemes/toric/weierstrass.py [hunk 3]
```diff
@@ -220,10 +220,10 @@ def j_invariant(polynomial, variables=None):
         ...
         ValueError: curve is singular and has no well-defined j-invariant
     """
-    (f, g) = WeierstrassForm(polynomial, variables)
-    disc = 4*f**3+27*g**2
+    f, g = WeierstrassForm(polynomial, variables)
+    disc = 4 * f**3 + 27 * g**2
```
Comment (mkoeppe) [inline src/sage/schemes/toric/weierstrass.py]:
```suggestion
    disc = 4*f**3 + 27*g**2
```
top-level only please

File: src/sage/schemes/toric/weierstrass.py [hunk 4]
```diff
@@ -286,7 +286,7 @@ def Newton_polytope_vars_coeffs(polynomial, variables):
         e = m.exponents()[0]
         v = tuple([e[i] for i in var_indices])
         m_red = m // prod(x**i for x, i in zip(variables, v))
-        result[v] = result.get(v, R.zero()) + c*m_red
+        result[v] = result.get(v, R.zero()) + c * m_red
```
Comment (mkoeppe) [inline src/sage/schemes/toric/weierstrass.py]:
```suggestion
        result[v] = result.get(v, R.zero()) + c*m_red
```
whitespace around top-level arith operators only please

File: src/sage/schemes/toric/weierstrass.py [hunk 5]
```diff
@@ -553,7 +553,7 @@ def _check_homogeneity(polynomial, variables, weights, total_weight=None):
         else:
             if weight_e != total_weight:
                 raise ValueError('the polynomial is not homogeneous with '
-                                 'weights '+str(weights))
+                                 'weights ' + str(weights))
```
Comment (mkoeppe) [inline src/sage/schemes/toric/weierstrass.py]:
```suggestion
                                 f'weights {weights}')
```

File: src/sage/schemes/toric/weierstrass.py [hunk 6]
```diff
@@ -599,7 +599,7 @@ def index(monomial):
     coeffs = {}
     for c, m in polynomial:
         i = index(m)
-        coeffs[i] = c*m + coeffs.pop(i, R.zero())
+        coeffs[i] = c * m + coeffs.pop(i, R.zero())
```
Comment (mkoeppe) [inline src/sage/schemes/toric/weierstrass.py]:
```suggestion
        coeffs[i] = c*m + coeffs.pop(i, R.zero())
```

File: src/sage/schemes/toric/weierstrass.py [hunk 7]
```diff
@@ -853,10 +853,10 @@ def _partial_discriminant(quadric, y0, y1=None):
         monomials = (quadric.parent().one(), y0, y0**2)
         variables = [y0]
     else:
-        monomials = (y1**2, y0*y1, y0**2)
+        monomials = (y1**2, y0 * y1, y0**2)
         variables = [y0, y1]
     c = _extract_coefficients(quadric, monomials, variables)
-    return c[1]**2 - 4*c[0]*c[2]
+    return c[1]**2 - 4 * c[0] * c[2]
```
Comment (mkoeppe) [inline src/sage/schemes/toric/weierstrass.py]:
```suggestion
    return c[1]**2 - 4*c[0]*c[2]
```

**Comment by fchapoton** [comment]
done, but let me respectfuly note once again that this goes against the pycodestyle implemented rules

**Comment by mkoeppe** [comment]
> done, but let me respectfuly note once again that this goes against the pycodestyle implemented rules

Thank you. It's too bad that pycodestyle does not implement PEP 8 faithfully. Do you know if Ruff has an opinion on this matter?

**Review by mkoeppe — APPROVED** [review]

**Comment by fchapoton** [comment]
Thanks. I think ruff is deliberatly not concerned with code-formatting, only with code enhancements.

**Pull request merged**